src/main/java/com/bandwidth/sqs/consumer/SqsConsumerManager.java

```diff
@@ -81,8 +104,28 @@ public synchronized void updateAllocatedInFlightRequests(SqsConsumer consumer, U
         }
     }
 
-    public void queueTask(Runnable task) {
-        threadPool.execute(task);
+    public void queueTask(Function<SqsMessage<T>, Completable> task, int priority, SqsConsumer<T> consumer) {
+
+        //queue a task that we want to run on the thread pool. If there are more tasks than threads
+        //the highest priority tasks will run first
+        threadPoolQueue.add(new PriorityTask<>(priority, consumer, (msg) -> {
+            Completable.defer(() -> task.apply(msg))
+                    .doFinally(taskPermits::release).subscribe();
+        }));
+
+        threadPool.execute(() -> {
```

Comment:
I think it's also possible to create a threadpool that internally uses a priority queue. This was not used here since it's currently possible for users to provide their own threadpool and share it with other components. This library would need to own the threadpool to do that. I might look into this more in the future.